Lib/collections/__init__.py

```diff
@@ -691,9 +691,12 @@ def update(self, iterable=None, /, **kwds):
         if iterable is not None:
             if isinstance(iterable, _collections_abc.Mapping):
                 if self:
-                    self_get = self.get
                     for elem, count in iterable.items():
-                        self[elem] = count + self_get(elem, 0)
+                        if elem in self:
+                            self[elem] += count
```

Comment:
```suggestion
                            self[elem] = self[elem] + count
```
About the performance: results can depend on the system, platform and on the input type (e.g. adding a python `int`, a `bool` or `numpy.int16` to the counter). I tested with `%timeit counter.update({'a': 1})`. Both our tests show performance impact is small.

Perhaps you can even improve a bit by replacing `self[elem] += count` with `self[elem] = count + self[elem]`. A microbenchmark on my system:
```
import dis

def h():
    self={'a': 0}
    elem='a'
    for jj in range(1_000_000):
            self[elem] += 1

h()            

print(dis.dis(h, adaptive=True))

def g():
    self={'a': 0}
    elem='a'
    for jj in range(1_000_000):
            self[elem] = self[elem] + 1

g()            

print(dis.dis(g, adaptive=True))

%timeit h()
%timeit g()
```
Output of the timeit statements:
```
42.2 ms ± 831 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
36.6 ms ± 226 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```